From 70c280094ccd0fc718f3aabfe56a91b6bee81efd Mon Sep 17 00:00:00 2001
From: Christian Hergert <chergert@redhat.com>
Date: Thu, 18 Apr 2019 21:52:49 -0700
Subject: [PATCH] utils: force exit host commands when flatpak-builder exits

If we are spawning applications on the host using the Development service,
then we want those commands to exit when the flatpak-builder process
exits, as can happen from Ctrl^C or kill().
---
 src/builder-flatpak-utils.h | 1 +
 src/builder-utils.c         | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/builder-flatpak-utils.h b/src/builder-flatpak-utils.h
index 4d878d98..86631002 100644
--- a/src/builder-flatpak-utils.h
+++ b/src/builder-flatpak-utils.h
@@ -33,6 +33,7 @@
 
 typedef enum {
   FLATPAK_HOST_COMMAND_FLAGS_CLEAR_ENV = 1 << 0,
+  FLATPAK_HOST_COMMAND_FLAGS_WATCH_BUS = 1 << 1,
 } FlatpakHostCommandFlags;
 
 typedef void (*FlatpakLoadUriProgress) (guint64 downloaded_bytes,
diff --git a/src/builder-utils.c b/src/builder-utils.c
index 37de88c9..54bba04c 100644
--- a/src/builder-utils.c
+++ b/src/builder-utils.c
@@ -1750,7 +1750,8 @@ builder_host_spawnv (GFile                *dir,
                                                                       argv,
                                                                       g_variant_builder_end (fd_builder),
                                                                       g_variant_builder_end (env_builder),
-                                                                      FLATPAK_HOST_COMMAND_FLAGS_CLEAR_ENV),
+                                                                      FLATPAK_HOST_COMMAND_FLAGS_CLEAR_ENV |
+                                                                      FLATPAK_HOST_COMMAND_FLAGS_WATCH_BUS),
                                                        G_VARIANT_TYPE ("(u)"),
                                                        G_DBUS_CALL_FLAGS_NONE, -1,
                                                        fd_list, NULL,
